ts/src/alpaca.ts

```diff
@@ -503,6 +804,48 @@ export default class alpaca extends Exchange {
         return this.parseTrades (symbolTrades, market, since, limit);
     }
 
+    async fetchMyTrades (symbol: Str = undefined, since: Int = undefined, limit: Int = undefined, params = {}): Promise<Trade[]> {
+        /**
+         * @method
+         * @name alpaca#fetchMyTrades
+         * @description fetch all trades made by the user
+         * @see https://docs.alpaca.markets/reference/getaccountactivities-2
+         * @param {string} symbol unified symbol of the market to fetch trades for
+         * @param {int} [since] timestamp in ms of the earliest trade to fetch
+         * @param {int} [limit] the maximum amount of trades to fetch
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @param {string} [params.until] timestamp in ms of the earliest trade to fetch
+         * @param {string} [params.page_token] page_token - used for paging
+         * @returns {Trade[]} a list of [trade structures]{@link https://docs.ccxt.com/#/?id=public-trades}
+         */
+        await this.loadMarkets ();
+        const request: Dict = {
+            'category': 'trade_activity',
+        };
+        let market = undefined;
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+        }
+        const until = this.safeInteger (params, 'until');
+        if (until !== undefined) {
+            params = this.omit (params, 'until');
+            request['until'] = this.iso8601 (until);
+        }
+        if (since !== undefined) {
+            request['after'] = this.iso8601 (since);
+        }
+        if (limit !== undefined) {
+            request['page_size'] = limit;
+        }
+        // add a page token if in params
+        const page_token = this.safeString (params, 'page_token');
+        if (page_token !== undefined) {
+            request['page_token'] = page_token;
+        }
```

Comment:
this is not needed, at the end we always extend `request` with `params` so this will be done automatically if the user provides something in `params`

Reply:
Cool. Removed.